morphia/src/test/java/org/mongodb/morphia/IndexHelperTest.java

```diff
@@ -109,13 +111,13 @@ public void calculateKeys() {
     @Test
     public void createIndex() {
         checkMinServerVersion(3.4);
-        MongoCollection<Document> collection = getDatabase().getCollection("indexes");
-        MappedClass mappedClass = getMorphia().getMapper().getMappedClass(IndexedClass.class);
+        String collectionName = getDs().getCollection(IndexedClass.class).getName();
+        MongoCollection<Document> collection = getDatabase().getCollection(collectionName);
+        Mapper mapper = getMorphia().getMapper();
 
-        indexHelper.createIndex(collection, mappedClass, false);
+        indexHelper.createIndex(collection, mapper.getMappedClass(IndexedClass.class), false);
         List<DBObject> indexInfo = getDs().getCollection(IndexedClass.class)
                                           .getIndexInfo();
-        assertEquals("Should have 5 indexes", 5, indexInfo.size());
```

Comment:
Well my initial thought was we'd catch any extra indexes in the loop but that won't catch missing indexes so I'll restore that check.